tests/fustat_programs/test_sha256.cairo

```diff
@@ -0,0 +1,86 @@
+%builtins range_check bitwise
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.cairo.common.alloc import alloc
+from src.sha import SHA256, sha256, HashUtils
+from src.utils import pow2alloc128
+
+func main{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}() {
+    alloc_locals;
+
+    let (sha256_ptr, sha256_ptr_start) = SHA256.init();
+    let (pow2_array) = pow2alloc128();
+    local length: felt;
+
+    %{
+        import random
+        import os
+
+        def generate_hex_array(N):
+            hex_array = [os.urandom(i).hex() for i in range(1, N + 1)]
+            return hex_array
+
+        preimages = generate_hex_array(150)
+        print(preimages)
+
+        ids.length = len(preimages)
+    %}
+
+    with sha256_ptr, pow2_array {
+        run_test(index=length - 1);
+    }
+
+    SHA256.finalize(sha256_start_ptr=sha256_ptr_start, sha256_end_ptr=sha256_ptr);
+
+    return ();
+}
+
+func run_test{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, sha256_ptr: felt*, pow2_array: felt*}(
+    index: felt
+) {
+    alloc_locals;
+
+    if (index == 0) {
+        return ();
+    }
+
+    let (input: felt*) = alloc();
+    local n_bytes: felt;
+    local expected: Uint256;
+    %{
+        import hashlib
+        # This function chunks from MSB to LSB
+        def hex_to_chunks_32(hex_string: str):
+            # Remove '0x' prefix if present
+            if hex_string.startswith(('0x', '0X')):
+                hex_string = hex_string[2:]
+
+            # if we have an odd number of characters, prepend a 0
+            if len(hex_string) % 2 == 1:
+                hex_string = '0' + hex_string
+
+            # Now split into 8-character (32-bit) chunks
+            chunks = [int(hex_string[i:i+8], 16) for i in range(0, len(hex_string), 8)]
+            return chunks
```

Comment:
Use [garaga.hints.io.bigint_split](https://github.com/keep-starknet-strange/garaga/blob/3956fffbcfef7e6762eeb59618fa74327fadc502/hydra/garaga/hints/io.py#L50)

It Works with bytes as well. 

`chunks = io.bigint_split(hasher.digest(), 8, 2**32)[::-1] `

Reply:
super neat these functions!

Reply:
test keeps breaking, as I have variable length inputs, and need to compute n_limbs which keeps going wrong.